app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -394,8 +420,312 @@ public void onTaskFailure(String result) {
+        return results.getRawCursor();
+    }
+
+    private Cursor makePeopleCursor(CharSequence email) throws SQLException {
+        if (email == null) {
+            email = "";
+        }
+        String[] pieces = TextUtils.split(email.toString(), ",");
+        String piece;
+        if (pieces.length == 0) {
+            piece = "";
+        } else {
+            piece = pieces[pieces.length - 1].trim();
+        }
+        return ((AndroidDatabaseResults) app
+                .getDao(Person.class)
+                .queryRaw(
+                        "SELECT rowid _id, * FROM people WHERE "
+                                + Person.ISBOT_FIELD + " = 0 AND "
+                                + Person.ISACTIVE_FIELD + " = 1 AND "
+                                + Person.EMAIL_FIELD
+                                + " LIKE ? ESCAPE '\\' ORDER BY "
+                                + Person.NAME_FIELD + " COLLATE NOCASE",
+                        DatabaseHelper.likeEscape(piece) + "%")
+                .closeableIterator().getRawResults()).getRawCursor();
+    }
+    public void switchToStream() {
+        removeEditTextErrors();
+        if (!isCurrentModeStream()) switchView();
+    }
+
+    public void switchToPrivate() {
+        removeEditTextErrors();
+        if (isCurrentModeStream()) switchView();
+    }
+
+    public boolean isCurrentModeStream() {
+        //The TextView is VISIBLE which means currently send to stream is on.
+        return (textView.getVisibility() == View.VISIBLE);
+    }
+
+    public void removeEditTextErrors() {
+        streamActv.setError(null);
+        topicActv.setError(null);
+        messageEt.setError(null);
+    }
+
+    public void switchView() {
+        if (isCurrentModeStream()) { //Person
+            togglePrivateStreamBtn.setImageDrawable(ContextCompat.getDrawable(this, R.drawable.ic_action_bullhorn));
+            tempStreamSave = topicActv.getText().toString();
+            topicActv.setText(null);
+            topicActv.setHint(R.string.hint_person);
+            topicActv.setAdapter(emailActvAdapter);
+            streamActv.setVisibility(View.GONE);
+            textView.setVisibility(View.GONE);
+        } else { //Stream
+            topicActv.setText(tempStreamSave);
+            togglePrivateStreamBtn.setImageDrawable(ContextCompat.getDrawable(this, R.drawable.ic_action_person));
+            streamActv.setEnabled(true);
+            topicActv.setHint(R.string.hint_subject);
+            streamActv.setHint(R.string.hint_stream);
+            streamActv.setVisibility(View.VISIBLE);
+            textView.setVisibility(View.VISIBLE);
+            topicActv.setVisibility(View.VISIBLE);
+            streamActv.setAdapter(streamActvAdapter);
+            topicActv.setAdapter(subjectActvAdapter);
+        }
+    }
+    String tempStreamSave = null;
+
+    @Override
+    public void clearChatBox(){
+        if(TextUtils.isEmpty(messageEt.getText())){
```

Comment:
Well yeah I did this so messageEt should not be null at that time, anyways I have added a null check as well!